### PR TITLE
Feature/get status covid by country endpoint

### PR DESCRIPTION
**Pull request by oscardiazencora**
-added new endpoint get status by country  in status controller
-refactor ICovidService

### CODE REVIEW

File: Services/CovidService.cs [hunk 1]
```diff
@@ -17,16 +18,33 @@ public CovidService(IConfiguration config)
...
+        /// <returns></returns>
+        public async Task<IEnumerable<CovidStatus>> GetStatusByCountryAsync(string countryCode)
         {
-            throw new System.NotImplementedException();
+            var query = await _baseUrl
+                .AppendPathSegment(RouteConstants.ByStatus)
+                .GetJsonAsync<List<CovidStatus>>()
+                .ConfigureAwait(false);
+
+            query = query.Where(p => p.Country.Equals(countryCode)).ToList();
```
Comment (yeseniamolinab) [inline Services/CovidService.cs]: Should we care about casing in this comparison?

**Review by yeseniamolinab — COMMENTED** [review]
I have a question about the `Equals` line, otherwise it looks good to me.

File: Services/CovidService.cs (hunk 1 above)
Comment (oscardiazencora) [reply Services/CovidService.cs]: good question , let me take a look at the api

**Review by oscardiazencora — COMMENTED** [review]
I just cast it to upper case, so that its transparent for the user

File: Services/CovidService.cs (hunk 1 above)
Comment (oscardiazencora) [reply Services/CovidService.cs]: I just updated the new change, by the way good question

**Review by yeseniamolinab — APPROVED** [review]

**Pull request merged**